src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -116,8 +118,40 @@ bool ParserSubquery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     }
     else if (ASTPtr explain_node; explain.parse(pos, explain_node, expected))
     {
-        /// Replace SELECT * FROM (EXPLAIN SELECT ...) with SELECT * FROM viewExplain(EXPLAIN SELECT ...)
-        result_node = buildSelectFromTableFunction(makeASTFunction("viewExplain", explain_node));
+        const auto & explain_query = explain_node->as<const ASTExplainQuery &>();
+
+        if (explain_query.getTableFunction() || explain_query.getTableOverride())
+            throw Exception("EXPLAIN in a subquery cannot have a table function or table override", ErrorCodes::BAD_ARGUMENTS);
+
+        /// Replace `SELECT * FROM (EXPLAIN <kind> <explain_settings> SELECT ...)`
```

Comment:
Comment looks incorrect, we replace ```(EXPLAIN <kind> <explain_settings> SELECT ...)``` into `SELECT * FROM viewExplain("<kind>", "<explain_settings>", SELECT ...)`.

Reply:
I expect that it should work even for scalar subqueries. If we have ```(SELECT (EXPLAIN ))``` it will be replaced into ```SELECT (SELECT * FROM viewExplain())```. Should we add some test on it ?

Reply:
In the most cases `EXPLAIN` returns more then one row, but if it doesn't, it can be used as scalar, e.g. `SELECT (EXPLAIN SYNTAX SELECT 1)` added to test.